pallets/ajuna-affiliates/src/impls/fee_handler.rs

```diff
@@ -1,21 +1,22 @@
 use super::*;
-use ajuna_primitives::fee_handler::FeeProvider;
+use ajuna_primitives::fee_handler::{DistributeFee, Payment};
+use frame_support::traits::Defensive;
 use sp_runtime::{traits::CheckedDiv, Saturating};
 
-impl<T: Config<I>, I: 'static> FeeProvider for Pallet<T, I> {
+impl<T: Config<I>, I: 'static> DistributeFee for Pallet<T, I> {
 	type AccountId = AccountIdFor<T>;
 	type FeeIdentifier = RuleIdentifierFor<T, I>;
-	type FeeCurrency = BalanceOf<T, I>;
-	type FeeOutput = Vec<(Self::FeeCurrency, Self::AccountId)>;
+	type Balance = BalanceOf<T, I>;
```

Comment:
I decided to go with the naming that substrate uses. Substrate uses `Currency/Fungibles` if the complex trait is meant, and simply `Balance` if it is about the scalar number type of the balance.